gapis/api/vulkan/frame_loop.go

```diff
@@ -779,14 +791,75 @@ func (f *frameLoop) buildStartEndStates(ctx context.Context, startState *api.Glo
 				f.renderPassToCreate[renderPass] = true
 			}
 
-			// TODO:  Recreate destroyed resources.
-		}
+		// Command Buffers
+		case *VkAllocateCommandBuffers:
+			vkCmd := cmd.(*VkAllocateCommandBuffers)
+			cmdBuffers := vkCmd.PCommandBuffers().MustRead(ctx, vkCmd, currentState, nil)
+			f.commandBufferToFree[cmdBuffers] = true
+			log.D(ctx, "Command buffer %v allcoated.", cmdBuffers)
+		case *VkFreeCommandBuffers:
+			vkCmd := cmd.(*VkFreeCommandBuffers)
+			cmdBufCount := vkCmd.CommandBufferCount()
+			cmdBufs := vkCmd.PCommandBuffers().Slice(0, uint64(cmdBufCount), currentState.MemoryLayout).MustRead(ctx, cmd, currentState, nil)
+			for _, cmdBuf := range cmdBufs {
+				// The command buffer deleted in this call was created during loop, no action needed.
+				log.D(ctx, "Command buffer %v freed", cmdBufs)
+				if _, ok := f.commandBufferToFree[cmdBuf]; ok {
+					delete(f.commandBufferToFree, cmdBuf)
+				} else { // The command buffer deleted in this call was not created during loop, need to back up it
+					f.commandBufferToAllocate[cmdBuf] = true
+				}
+			}
+		case *VkBeginCommandBuffer:
```

Comment:
Tiny "nit", but the other commands all have a single empty line between the cases.